packages/typespec-ts/src/modular/buildRootIndex.ts

```diff
@@ -5,83 +5,90 @@ import { Client } from "./modularCodeModel.js";
 export function buildRootIndex(
   client: Client,
   project: Project,
-  srcPath: string
+  rootIndexFile: SourceFile,
+  srcPath: string,
+  subfolder: string,
+  isLast: boolean
 ) {
   const clientName = `${getClientName(client)}Client`;
-  const clientFile = project.getSourceFile(`${srcPath}/src/${clientName}.ts`);
+  const clientFile = project.getSourceFile(
+    `${srcPath}/src/${subfolder !== "" ? subfolder + "/" : ""}${clientName}.ts`
+  );
 
   if (!clientFile) {
     throw new Error(
       `Couldn't find client file: ${srcPath}/src/${clientName}.ts`
     );
   }
 
-  const file = project.createSourceFile(`${srcPath}/src/index.ts`, "", {
-    overwrite: true
-  });
+  exportClassicalClient(client, rootIndexFile, subfolder);
 
-  exportModels(file, srcPath);
-  exportOptionsInterfaces(client, file, srcPath);
-  exportClassicalClient(client, file);
+  if (isLast) {
+    // exportApiIndex(rootIndexFile, srcPath);
+    exportModels(rootIndexFile, project, srcPath);
```

Comment:
We need isLast, to build the models, because those the models are just ready by that time.